modules/shortcodes/crowdsignal.php

```diff
@@ -1,610 +1,686 @@
+						$content = str_replace( $match[0], " [crowdsignal poll=$id]", $content );
+						/** This action is documented in modules/shortcodes/youtube.php */
+						do_action( 'jetpack_embed_to_shortcode', 'crowdsignal', $id );
+					}
 				}
 			}
-		}
 
-		return $content;
-	}
-
-	/**
-	 * Shortcode for polldadddy
-	 * [crowdsignal poll|survey|rating="123456"]
-	 */
-	function crowdsignal_shortcode( $atts ) {
-		global $post;
-		global $content_width;
+			return $content;
+		}
 
 		/**
-		 * Variables extracted from $atts.
+		 * Shortcode for polldadddy
+		 * [crowdsignal poll|survey|rating="123456"]
 		 *
-		 * @var string $survey
-		 * @var string $link_text
-		 * @var string $poll
-		 * @var string $rating
-		 * @var string $unique_id
-		 * @var string $item_id
-		 * @var string $title
-		 * @var string $permalink
-		 * @var int $cb
-		 * @var string $type
-		 * @var string $body
-		 * @var string $button
-		 * @var string $text_color
-		 * @var string $back_color
-		 * @var string $align
-		 * @var string $style
-		 * @var int $width
-		 * @var int $height
-		 * @var int $delay
-		 * @var string $visit
-		 * @var string $domain
-		 * @var string $id
+		 * @param array $atts Shortcode attributes.
 		 */
-		extract( shortcode_atts( array(
-			'survey'     => null,
-			'link_text'  => 'Take Our Survey',
-			'poll'       => 'empty',
-			'rating'     => 'empty',
-			'unique_id'  => null,
-			'item_id'    => null,
-			'title'      => null,
-			'permalink'  => null,
-			'cb'         => 0,
-			'type'       => 'button',
-			'body'       => '',
-			'button'     => '',
-			'text_color' => '000000',
-			'back_color' => 'FFFFFF',
-			'align'      => '',
-			'style'      => '',
-			'width'      => $content_width,
-			'height'     => floor( $content_width * 3 / 4 ),
-			'delay'      => 100,
-			'visit'      => 'single',
-			'domain'     => '',
-			'id'         => '',
-		), $atts, 'crowdsignal' ) );
-
-		if ( ! is_array( $atts ) ) {
-			return '<!-- Crowdsignal shortcode passed invalid attributes -->';
-		}
-
-		$inline          = ! in_the_loop();
-		$no_script       = false;
-		$infinite_scroll = false;
-
-		if ( is_home() && current_theme_supports( 'infinite-scroll' ) ) {
-			$infinite_scroll = true;
-		}
-
-		if ( defined( 'PADPRESS_LOADED' ) ) {
-			$inline = true;
-		}
-
-		if ( function_exists( 'get_option' ) && get_option( 'polldaddy_load_poll_inline' ) ) {
-			$inline = true;
-		}
-
-		if ( is_feed() || ( defined( 'DOING_AJAX' ) && ! $infinite_scroll ) ) {
-			$no_script = false;
-		}
-
-		self::$add_script = $infinite_scroll;
-
-		if ( intval( $rating ) > 0 && ! $no_script ) { //rating embed
-
-			if ( empty( $unique_id ) ) {
-				$unique_id = is_page() ? 'wp-page-' . $post->ID : 'wp-post-' . $post->ID;
+		public function crowdsignal_shortcode( $atts ) {
+			global $post;
+			global $content_width;
+
+			$attributes = shortcode_atts(
+				array(
+					'survey'     => null,
+					'link_text'  => esc_html__( 'Take Our Survey', 'jetpack' ),
```

Comment:
Should this use "jetpack" as it'll be on WordPress.com?

Reply:
No need. WordPress.com handles the `jetpack` textdomain well.